fcrepo4-exts/fcrepo-camel
# Fedora 5 compatibility

***Upgrades fcrepo-camel for fcrepo5 compatibility***
* * *

**JIRA Ticket**: https://jira.duraspace.org/browse/FCREPO-2965

# What does this Pull Request do?
* Refactors the `EventProcessor` class so that it translates fcrepo5-style bodies into the expected camel headers
* Uses fcrepo5, enables authn/authz for integration tests
* Refactors tests that unnecessarily inspected the contents of message bodies, in order to hold the door open for backwards compatibility with fcrepo4
* Removes the `LdnProcessor` and test.

# What's new?
The changes in the PR allow translation of fcrepo5 messages into the expected fcrepo camel headers.  Since the message bodies semantically differ between fcrepo4 and fcrepo5, messages from the two are not compatible.  This PR addresses fcrepo5 only, at the cost of losing compatibility with fcrepo4.  A future PR shall explore eanabling backward compatibility such that fcrepo-camel can handle both kinds of messages.

Additionally, the `LdnProcessor` has been removed.  The purpose of this processor is not entirely clear, and it is not part of `fcrepo-camel-toolbox`.  Some thought would be necessary in order to figure out a plausible way to retain `LdnProcessor` in a way that is mutually compatible with fcrepo4 and fcrepo5.  This PR punts and removes a seemingly little-used processor for the sake of expediency.


# How should this be tested?
Build, and see if tests pass.  If the reviewer has time and camel routes, try dropping this `fcrepo-camel` in and see what happens.


# Interested parties
@peichman-umd, @fcrepo4/committers
